IHP/Postgres/Interval.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-|
+Module: IHP.Postgres.Interval
+Description: Adds support for the Postgres Interval type
+Copyright: (c) digitally induced GmbH, 2020
+-}
+module IHP.Postgres.Interval where
+
+import BasicPrelude
+
+import Database.PostgreSQL.Simple.ToField
+import Database.PostgreSQL.Simple.FromField
+import qualified Database.PostgreSQL.Simple.TypeInfo.Static as TI
+import Database.PostgreSQL.Simple.TypeInfo.Macro as TI
+import Data.Attoparsec.ByteString.Char8 as Attoparsec
+import Data.String.Conversions (cs)
+import Data.Aeson
+
+import IHP.Postgres.TimeParser (PGInterval(..))
+
+instance FromField PGInterval where
+    fromField f v =
+        if typeOid f /= $(inlineTypoid TI.interval)
+        then returnError Incompatible f ""
+        else case v of
+               Nothing -> returnError UnexpectedNull f ""
+               Just bs -> case parseOnly pPGInterval bs of
+                     Left  err -> returnError ConversionFailed f err
+                     Right val -> pure val
+
+pPGInterval = do
+    bs <- takeByteString
+    pure (PGInterval bs)
+
+instance ToField PGInterval where
+    toField (PGInterval interval) = toField (interval)
+
+instance FromJSON PGInterval where
+     parseJSON (Object v) = (PGInterval . encodeUtf8) <$> ((v .: "interval"))
+     parseJSON _ = mzero
+
+instance ToJSON PGInterval where
+    toJSON (PGInterval pgInterval) = object [ "interval" .= (tshow pgInterval)]
```

Comment:
```suggestion
    toJSON (PGInterval pgInterval) = object [ "interval" .= (tshow pgInterval) ]
```

Reply:
After further testing I think the toJSON method should actually just return a string of the Postgres interval rather than an object. 

if you have a data structure:
data Activity = Activity {duration :: PGInterval}

in Client you would rather access this as:

```
const activity = useQuerySingleResult(query("activities"))
//interval string
<p> Interval: {activity.duration} </p>

// unnecessary object {activity.duration.interval}
```

Reply:
agree 👍